lahja/endpoint.py

```diff
@@ -129,6 +174,65 @@ def connect_no_wait(self, loop: Optional[asyncio.AbstractEventLoop] = None) -> N
 
             self._process_item(item, config)
 
+    def _create_external_api(self, ipc_path: pathlib.Path) -> None:
+
+        receiver = EndpointConnector(self)
+
+        ConnectorManager = get_connector_manager()
+
+        ConnectorManager.register('get_connector', callable=lambda: receiver)  # type: ignore
+
+        manager = ConnectorManager(address=str(ipc_path))  # type: ignore
+        server = manager.get_server()   # type: ignore
+        threading.Thread(target=server.serve_forever, daemon=True).start()
+
+    def connect_to_endpoints_blocking(self, *endpoints: ConnectionConfig, timeout: int=30) -> None:
```

Comment:
I think you might not need this method, and transitively you might not need `wait_for_path_blocking`.

Every caller of `connect_to_endpoints_blocking` has an EventLoop, and could instead call:

```
loop.run_until_complete(endpoint.connect_to_endpoints(...))
```

This would even give a small performance boost, because `connect_to_endpoints` works in parallel.

Reply:
Ha! That's a really good point! Eagle eyes!

Reply:
Ah wait. I tried this and run into problems. Because, we could only do `loop.run_until_complete` if this loop did not start already. Which is an assumption that doesn't hold because you should be able to connect to further endpoints at any time.

Reply:
Oops :( Thanks for trying it! I should have tried it before making the suggestion. There might be a solution like:

```
if self._loop.running():
   ...
...
```

but that doesn't sound any better than leaving this method as-is :+1: